File: src/Entity/RessourceEducative.php
<?php

namespace App\Entity;

use App\Repository\RessourceEducativeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: RessourceEducativeRepository::class)]
class RessourceEducative
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $titre = null;

    #[ORM\Column(type: Types::TEXT)]
    private ?string $contenu = null;

    #[ORM\Column(length: 255)]
    private ?string $auteur = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $datePublication = null;

    #[ORM\Column(length: 255)]
    private ?string $image = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $video = null;

    #[ORM\ManyToOne(targetEntity: CategorieRessource::class, inversedBy: "ressources")]
    #[ORM\JoinColumn(nullable: false)]
    private ?CategorieRessource $categorie = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): static
    {
        $this->titre = $titre;

        return $this;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): static
    {
        $this->contenu = $contenu;

        return $this;
    }

    public function getAuteur(): ?string
    {
        return $this->auteur;
    }

    public function setAuteur(string $auteur): static
    {
        $this->auteur = $auteur;

        return $this;
    }

    public function getDatePublication(): ?\DateTimeInterface
    {
        return $this->datePublication;
    }

    public function setDatePublication(\DateTimeInterface $datePublication): static
    {
        $this->datePublication = $datePublication;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): static
    {
        $this->image = $image;

        return $this;
    }

    public function getVideo(): ?string
    {
        return $this->video;
    }

    public function setVideo(?string $video): static
    {
        $this->video = $video;

        return $this;
    }

    public function getCategorie(): ?CategorieRessource
    {
        return $this->categorie;
    }

    public function setCategorie(?CategorieRessource $categorie): static
    {
        $this->categorie = $categorie;

        return $this;
    }
}

File: src/main/java/models/RessourceEducative.java
package models;

import java.time.LocalDateTime;

public class RessourceEducative {
    private int id;
    private String titre;
    private String contenu;
    private String auteur;
    private String image;
    private String video;
    private LocalDateTime datePublication;
    private CategorieRessource categorie;  // Lien avec CategorieRessource

    public RessourceEducative(String titre, String contenu, String auteur, String image, String video, LocalDateTime datePublication, CategorieRessource categorie) {
        this.titre = titre;
        this.contenu = contenu;
        this.auteur = auteur;
        this.image = image;
        this.video = video;
        this.datePublication = datePublication;
        this.categorie = categorie;
        this.id = 0;
    }

    public RessourceEducative() {
    }
    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public String getAuteur() {
        return auteur;
    }

    public void setAuteur(String auteur) {
        this.auteur = auteur;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getVideo() {
        return video;
    }

    public void setVideo(String video) {
        this.video = video;
    }

    public LocalDateTime getDatePublication() {
        return datePublication;
    }

    public void setDatePublication(LocalDateTime datePublication) {
        this.datePublication = datePublication;
    }

    public CategorieRessource getCategorie() {
        return categorie;
    }

    public void setCategorie(CategorieRessource categorie) {
        this.categorie = categorie;
    }
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

}
